internal/models/transformer_test.go

```diff
@@ -7,15 +7,82 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-var examplePubkey = Pubkey{
+var lzapKeyED = Pubkey{
 	Name: "lzap-ed25519-2021",
 	Body: "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEhnn80ZywmjeBFFOGm+cm+5HUwm62qTVnjKlOdYFLHN lzap",
 }
 
-func TestPubkeyFingerprintGeneration(t *testing.T) {
-	pk := examplePubkey
-	err := Transform(context.Background(), &pk)
-	if assert.NoError(t, err) {
-		assert.Equal(t, "SHA256:gL/y6MvNmJ8jDXtsL/oMmK8jUuIefN39BBuvYw/Rndk", pk.Fingerprint, "fingerprint must be set")
+var lzapKeyRSA = Pubkey{
+	Name: "lzap-rsa-2013",
+	Body: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC8w6DONv1qn3IdgxSpkYOClq7oe7davWFqKVHPbLoS6+dF" +
+		"Inru7gdEO5byhTih6+PwRhHv/b1I+Mtt5MDZ8Sv7XFYpX/3P/u5zQiy1PkMSFSz0brRRUfEQxhXLW97FJa7l+bej2HJ" +
+		"Dt7f9Gvcj+d/fNWC9Z58/GX11kWk4SIXaKotkN+kWn54xGGS7Zvtm86fP59Srt6wlklSsG8mZBF7jVUjyhAgm/V5gDF" +
+		"b2/6jfiwSb2HyJ9/NbhLkWNdwrvpdGZqQlYhnwTfEZdpwizW/Mj3MxP5O31HN45aE0wog0UeWY4gvTl4Ogb6kescizA" +
+		"M6pCff3RBslbFxLdOO7cR17 lzap-2013",
+}
+
+func TestFingerprintGeneration(t *testing.T) {
+	type test struct {
+		name        string
+		pubkey      Pubkey
+		fingerprint string
+	}
+
+	tests := []test{
+		{"ed25519", lzapKeyED, "gL/y6MvNmJ8jDXtsL/oMmK8jUuIefN39BBuvYw/Rndk="},
+		{"rsa", lzapKeyRSA, "ENShRe/0uDLSw9c+7tc9PxkD/p4blyB/DTgBSIyTAJY="},
+	}
+
+	for _, td := range tests {
+		t.Run(td.name, func(t *testing.T) {
+			err := Transform(context.Background(), &td.pubkey)
+			assert.NoError(t, err)
+			assert.Equal(t, td.fingerprint, td.pubkey.Fingerprint,
+				"%s fingerprint %s does not match %s", td.name, td.pubkey.Fingerprint, td.fingerprint)
+		})
+	}
+}
+
+func TestFingerprintLegacyGeneration(t *testing.T) {
+	type test struct {
+		name        string
+		pubkey      Pubkey
+		fingerprint string
+	}
+
+	tests := []test{
+		{"ed25519", lzapKeyED, "ee:f1:d4:62:99:ab:17:d9:3b:00:66:62:32:b2:55:9e"},
+		{"rsa", lzapKeyRSA, "89:c5:99:b5:33:48:1c:84:be:da:cb:97:45:b0:4a:ee"},
+	}
+
+	for _, td := range tests {
+		t.Run(td.name, func(t *testing.T) {
+			err := Transform(context.Background(), &td.pubkey)
+			assert.NoError(t, err)
+			assert.Equal(t, td.fingerprint, td.pubkey.FingerprintLegacy,
+				"%s fingerprint %s does not match %s", td.name, td.pubkey.FingerprintLegacy, td.fingerprint)
+		})
+	}
+}
+
+func TestFingerprintPEMGeneration(t *testing.T) {
+	type test struct {
+		name        string
+		pubkey      Pubkey
+		fingerprint string
+	}
+
+	tests := []test{
+		{"ed25519", lzapKeyED, "e3:8d:76:a4:f2:78:29:f5:6d:0b:95:5c:e9:80:47:85"},
+		{"rsa", lzapKeyRSA, "c4:ba:72:45:16:a9:2c:39:c3:99:8d:e7:16:01:9c:77"},
+	}
+
+	for _, td := range tests {
+		t.Run(td.name, func(t *testing.T) {
+			err := Transform(context.Background(), &td.pubkey)
+			assert.NoError(t, err)
+			assert.Equal(t, td.fingerprint, td.pubkey.FingerprintPEM,
+				"%s fingerprint %s does not match %s", td.name, td.pubkey.FingerprintPEM, td.fingerprint)
+		})
 	}
 }
```

Comment:
This is a complete test of all three methods on both supported SSH key types. I tried also to add DSA keys but they do not appear to be supported by the Go crypto library so I think we will only support the two.